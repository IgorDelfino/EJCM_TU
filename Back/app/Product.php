<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Product extends Model
{

    /* Compras que contém este produto */
    public function purchases() {
        return $this->belongsToMany('App\Purchase', 'carts')->withPivot('quantity');
    }

    /* Função que atualiza o estoque de produtos */
    public function updateStock($quantity) {
        if ($quantity > $this->stock) {
            return false;
        } else {
            $this->stock -= $quantity;
            return true;
        }
    }

}
