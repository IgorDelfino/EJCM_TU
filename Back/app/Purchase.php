<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Purchase extends Model
{
    public function createPurchase(Request $request){
        
        $purchase = new Purchase;

        $purchase->date = $request->date;
        $purchase->total_price = $request->total_price;

        $purchase->save();

        return json($purchase);
    }
}
