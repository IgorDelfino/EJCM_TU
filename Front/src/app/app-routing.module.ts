import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { ProdutoComponent } from './produto/produto.component';


const routes: Routes = [{ path: 'cadastro', component: ProdutoComponent }];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
