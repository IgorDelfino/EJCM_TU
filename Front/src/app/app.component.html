<!DOCTYPE html>
<html lang="en" style="margin-top:0px;">

<head>
    <link href = "./app.component.scss" rel="style">
    <link href="https://fonts.googleapis.com/css2?family=Qwigley&display=swap" rel="stylesheet">
    <meta charset = "utf-8"/>
    <title>TU Paper</title>
</head>
    
    <body style=" margin-top: 0px; margin-left: 35px; margin-right:0px;">
             <router-outlet></router-outlet>        
    </body>
</html>
