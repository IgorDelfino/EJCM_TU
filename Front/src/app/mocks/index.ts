export const validUser = {
    name: '00132126',
    email: 'fakeemail@gmail.com',
    password: '123456'
};

export const blankUser = {
    name: '',
    email: '',
    password: ''
};