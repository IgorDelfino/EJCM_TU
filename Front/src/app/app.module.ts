import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { ProdutoComponent } from './produto/produto.component';
import { CadastroComponent } from './cadastro/cadastro.component';

@NgModule({
  declarations: [
    AppComponent,
    ProdutoComponent,
    CadastroComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
