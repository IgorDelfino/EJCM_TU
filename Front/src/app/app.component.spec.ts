import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { fakeAsync, tick, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA, ÉµivyEnabled as ivyEnabled } from '@angular/core';



// describe('AppComponent', () => {
//   beforeEach(async(() => {
//     TestBed.configureTestingModule({
//       imports: [
//         RouterTestingModule
//       ],
//       declarations: [
//         AppComponent
//       ],
//     }).compileComponents();
//   }));

//   it('should create the app', () => {
//     const fixture = TestBed.createComponent(AppComponent);
//     const app = fixture.componentInstance;
//     expect(app).toBeTruthy();
//   });

//   it(`should have as title 'TUPaper'`, () => {
//     const fixture = TestBed.createComponent(AppComponent);
//     const app = fixture.componentInstance;
//     expect(app.title).toEqual('TUPaper');
//   });

//   it('should render title', () => {
//     const fixture = TestBed.createComponent(AppComponent);
//     fixture.detectChanges();
//     const compiled = fixture.nativeElement;
//     expect(compiled.querySelector('.content span').textContent).toContain('TUPaper app is running!');
//   });
// });


describe('Teste para UI', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
  }).compileComponents();
  }));
  // let component: AppComponent;
    it('Deve renderizar mensagem no h1', async(() => {
      const fixture = TestBed.createComponent(AppComponent);
      const compiled = fixture.debugElement.nativeElement;
      expect(compiled.querySelector('h1').textContent).toContain('helloworld');
  }))})

