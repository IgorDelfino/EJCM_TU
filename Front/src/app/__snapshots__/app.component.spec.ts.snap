// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste para UI Deve tirar o snapshot 1`] = `
<app-root
  title={[Function String]}
>
  <html
    lang="en"
    style="margin-top:0px;"
  >
    <head>
      <link
        href="./app.component.scss"
        rel="style"
      />
      <link
        href="https://fonts.googleapis.com/css2?family=Qwigley&display=swap"
        rel="stylesheet"
      />
      <meta
        charset="utf-8"
      />
      <title>
        TU Paper
      </title>
    </head>
    <div
      class="header"
    >
      <h1
        class="titulo"
      >
        TU Paper
      </h1>
    </div>
    <body
      style="display:flex; margin-top: 0px; margin-left: 35px; margin-right:0px;"
    >
      <div
        class="content"
      >
        <app-produto />
      </div>
    </body>
  </html>
</app-root>
`;
